### ISSUE TITLE
Error on install: fails on npm run build step

### ISSUE THREAD

**Issue by corbmanj** [stackgl/shader-school#118]
Here is the full output when I enter the command:
sudo npm install -g shader-school
npm WARN deprecated rework-inline@0.2.0: deprecated in favor of rework-import
 
> fsevents@0.2.1 install /usr/local/lib/node_modules/shader-school/node_modules/glslify-live/node_modules/chokidar/node_modules/fsevents
> node-gyp rebuild

shell-init: error retrieving current directory: getcwd: cannot access parent directories: Permission denied
sh: node-gyp: command not found
npm WARN optional dep failed, continuing fsevents@0.2.1
\
> fsevents@0.3.5 install /usr/local/lib/node_modules/shader-school/node_modules/beefy/node_modules/chokidar/node_modules/fsevents
> node-gyp rebuild

shell-init: error retrieving current directory: getcwd: cannot access parent directories: Permission denied
sh: node-gyp: command not found
npm WARN optional dep failed, continuing fsevents@0.3.5
 
> canvas@1.2.1 install /usr/local/lib/node_modules/shader-school/node_modules/gl-axes/node_modules/vectorize-text/node_modules/canvas-browserify/node_modules/canvas
> node-gyp rebuild

shell-init: error retrieving current directory: getcwd: cannot access parent directories: Permission denied
sh: node-gyp: command not found
npm WARN optional dep failed, continuing canvas@1.2.1
npm WARN engine pngjs@0.4.0: wanted: {"node":"0.8.x"} (current: {"node":"0.10.33","npm":"2.5.0"})
-
> stanford-dragon@1.1.1 postinstall /usr/local/lib/node_modules/shader-school/node_modules/stanford-dragon
> npm run build

shell-init: error retrieving current directory: getcwd: cannot access parent directories: Permission denied
sh: npm: command not found
npm ERR! Darwin 13.4.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "-g" "shader-school"
npm ERR! node v0.10.33
npm ERR! npm  v2.5.0
npm ERR! code ELIFECYCLE

npm ERR! stanford-dragon@1.1.1 postinstall: `npm run build`
npm ERR! Exit status 127
npm ERR! 
npm ERR! Failed at the stanford-dragon@1.1.1 postinstall script 'npm run build'.
npm ERR! This is most likely a problem with the stanford-dragon package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get their info via:
npm ERR!     npm owner ls stanford-dragon
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/corbmanj/npm-debug.log

**Comment by donmccurdy** [comment]
I ran into some [issues of my own](https://gist.github.com/donmccurdy/0046845a3f13b1609b0b) while trying to run `npm install -g shader-school`. It looks like yours are a bit different, possibly the permissions are wrong on the NPM directories? That can happen if NPM has been run with sudo in the past. If this is the case, you might be able to fix it by changing your permissions. Just guessing here:

```bash
sudo chown -R [your-username] /usr/local/lib/node_modules
```

If that doesn't work, it's relatively easy to install a new Node environment with [NVM](https://github.com/creationix/nvm). I switched from system Node to v0.10.31 and that fixed my problem. Might help in your case as well!
